csum-file.c

```diff
@@ -132,10 +139,14 @@ struct hashfile *hashfd_check(const char *name)
 		die_errno("unable to open '%s'", name);
 	f = hashfd(sink, name);
 	f->check_fd = check;
+	f->check_buffer = xmalloc(f->buffer_len);
```

Comment:
Can we be certain that there is no pre-existing `f->check_buffer` that now would leak?

Reply:
We are creating it from scratch, and `hashfd` initializes the struct with a `NULL` there.